src/app/backend/eventscommon.go

```diff
@@ -0,0 +1,145 @@
+// limitations under the License.
+
+package main
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"log"
+	"strings"
+)
+
+// Partial string to correctly filter warning events.
+// Has to be lower case for correct case insensitive comparison.
+const FAILED_REASON_PARTIAL = "failed"
+
+// Contains basic information about event related to a pod
+type PodEvent struct {
+	// Short, machine understandable string that gives the reason
+	// for this event being generated.
+	Reason string `json:"reason"`
+
+	// A human-readable description of the status of related pod.
+	Message string `json:"message"`
+}
+
+// Returns Pod Events of type warning from given list of pods
+// TODO(floreks) : Import and use Set instead of custom function to get rid of duplicates
+func GetPodsEventWarnings(client *client.Client, pods []api.Pod) (result []PodEvent, err error) {
+	for _, pod := range pods {
+		if !isRunningOrSucceeded(pod) {
+			log.Printf("Getting warning events from pod: %s", pod.Name)
+			events, err := GetPodEvents(client, pod)
+
+			if err != nil {
+				return nil, err
+			}
+
+			result = getPodsEventWarnings(events)
+		}
+	}
+
+	return removeDuplicates(result), nil
+}
+
+// Returns list of Pod Event model objects based on kubernetes API event list object
+// Event list object is filtered to get only events of type warning.
+func getPodsEventWarnings(eventList *api.EventList) []PodEvent {
+	result := make([]PodEvent, 0)
+
+	var events []api.Event;
+	if isTypeFilled(eventList.Items) {
+		events = filterEventsByType(eventList.Items, api.EventTypeWarning)
+	} else {
+		events = filterEventsByReason(eventList.Items, FAILED_REASON_PARTIAL)
+	}
+
+	for _, event := range events {
+		result = append(result, PodEvent{
+			Message: event.Message,
+			Reason:  event.Reason,
+		})
+	}
+
+	return result
+}
+
+// Filters kubernetes API event objects based on event type. Empty string will return all events.
+func filterEventsByType(events []api.Event, eventType string) []api.Event {
+	if len(eventType) == 0 || len(events) == 0 {
+		return events
+	}
+
+	result := make([]api.Event, 0)
+	for _, event := range events {
+		if event.Type == eventType {
+			result = append(result, event)
+		}
+	}
+
+	return result
+}
+
+// Filters kubernetes API event objects based on reason property. Empty string will return
+// all events.
+func filterEventsByReason(events []api.Event, partial string) []api.Event {
+	if len(partial) == 0 || len(events) == 0 {
+		return events
+	}
+
+	result := make([]api.Event, 0)
+	for _, event := range events {
+		if strings.Contains(strings.ToLower(event.Reason), partial) {
+			result = append(result, event)
+		}
+	}
+
+	return result
+}
+
+// Returns true if type of event is filled, false otherwise.
+// This is needed as some older versions of kubernetes do not have Type property filled.
+func isTypeFilled(events []api.Event) bool {
+	for _, event := range events {
+		return len(event.Type) != 0
```

Comment:
Hmm... This is equivalent to checking for only first event's type. How about for every event you check whether it has type or not. This is the way we should do this, because, I imagine, that some events have type and some not.